app/models/manageiq/providers/openshift/container_manager/refresher.rb

```diff
@@ -23,24 +23,29 @@ def fetch_hawk_inv(ems)
         return nil
       end
 
-      def parse_legacy_inventory(ems)
+      # Full refresh. Collecting immediately. Don't have separate Collector classes.
+      def collect_inventory_for_targets(ems, _targets)
         request_entities = OPENSHIFT_ENTITIES.dup
         request_entities << {:name => 'images'} if refresher_options.get_container_images
 
-        kube_entities = ems.with_provider_connection(:service => KUBERNETES_EMS_TYPE) do |kubeclient|
+        kube_inventory = ems.with_provider_connection(:service => KUBERNETES_EMS_TYPE) do |kubeclient|
           fetch_entities(kubeclient, KUBERNETES_ENTITIES)
         end
-        openshift_entities = ems.with_provider_connection do |openshift_client|
+        openshift_inventory = ems.with_provider_connection do |openshift_client|
           fetch_entities(openshift_client, request_entities)
         end
-        entities = openshift_entities.merge(kube_entities)
-        entities["additional_attributes"] = fetch_hawk_inv(ems) || {}
-        EmsRefresh.log_inv_debug_trace(entities, "inv_hash:")
 
+        inventory = openshift_inventory.merge(kube_inventory)
+        inventory["additional_attributes"] = fetch_hawk_inv(ems) || {}
+        EmsRefresh.log_inv_debug_trace(inventory, "inv_hash:")
+        [[ems, inventory]]
+      end
+
+      def parse_targeted_inventory(ems, _target_is_ems, inventory)
```

Comment:
context: the 2 new methods are both called from EmsRefresherMixin:
https://github.com/durandom/manageiq/blob/f9b9bd5fbc/app/models/ems_refresh/refreshers/ems_refresher_mixin.rb#L71-L94
(https://github.com/ManageIQ/manageiq/pull/13728 laid the framework but most of its explanation doesn't apply as I'm not using "Collector", "Target" classes, just defining these 2 methods.)

Previously we were using the naive default implementations where `collect_inventory_for_targets` did nothing
and `parse_targeted_inventory` called `parse_legacy_inventory` that both collected and parsed.